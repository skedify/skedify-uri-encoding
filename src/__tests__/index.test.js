import atob from 'atob';
import btoa from 'btoa';

import { encodeURIParameters, decodeURIParameters } from '../index';

if (!window.atob || !window.btoa) {
  window.atob = atob;
  window.btoa = btoa;
}

const exampleObject = {
  aKey: 'aValue',
};

describe('skedify-uri-encoding', () => {

  test('should return correct hash when no object is given', () => {
    expect(encodeURIParameters()).toMatchSnapshot();
  });

  test('should return correct hash when empty object is given', () => {
    expect(encodeURIParameters({})).toMatchSnapshot();
  });

  test('should return correct hash when an object is given', () => {
    expect(encodeURIParameters(exampleObject)).toMatchSnapshot();
  });

  test('should return correct object when string e30f is given', () => {
    expect(decodeURIParameters('e30f')).toMatchSnapshot();
  });

  test('should return correct object when string is given', () => {
    expect(decodeURIParameters('eyJhS2V5IjoiYVZhbHVlIn0f')).toMatchSnapshot();
  });

});
