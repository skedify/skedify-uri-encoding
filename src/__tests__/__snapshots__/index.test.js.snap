// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skedify-uri-encoding should return correct hash when an object is given 1`] = `"eyJhS2V5IjoiYVZhbHVlIn0f"`;

exports[`skedify-uri-encoding should return correct hash when empty object is given 1`] = `"e30f"`;

exports[`skedify-uri-encoding should return correct hash when no object is given 1`] = `"dW5kZWZpbmVkd"`;

exports[`skedify-uri-encoding should return correct object when string e30f is given 1`] = `Object {}`;

exports[`skedify-uri-encoding should return correct object when string is given 1`] = `
Object {
  "aKey": "aValue",
}
`;
